### ISSUE TITLE
Update clang and clang-format on CI?

### ISSUE THREAD

**Issue by dcaballe** [google/iree#9154]
It looks like the clang-format version used in CI is not very up-to-date and sometimes there are discrepancies between that version and more recent ones that we may use in development setups. The clang compiler version seems to be relatively old as well (6.0), which can be a problem when we have to deal with warnings/errors that may not be reproducible with more recent versions of the compiler. It would be great if we could update both tools in CI to a more recent (13?) version.

**Comment by GMNGeoffrey** [comment]
These are pretty separate decisions. With clang-format, we basically just have to pick one and pin it. The choice is pretty arbitrary, but if we want to do a later one, that's fine with me. With clang, we deliberately build with the oldest version of clang that we say we support. If we want to drop support for clang 6 explicitly, then we can have that conversation. It's clang-6 because that's the highest available for Ubuntu LTS release 18.04. Something I would like to do is have the latest version of clang in some builder *as well* so we get all the latest features covered

**Comment by dcaballe** [comment]
Keeping some instances of clang 6 and moving some of them to more recent versions sound great to me!

**Comment by GMNGeoffrey** [comment]
We've upgrade to clang-9 as our lowest supported version. I've filed https://github.com/google/iree/issues/9468 for also building with later clang versions on the CI. The clang-format version used is clang-format-9 as well: https://github.com/google/iree/blob/6a4c9df93f/.github/workflows/lint.yml#L97

Would you like to propose that we update it to a later version? No matter what version we use, there will be slight difference from whatever version someone has locally, unfortunately

**Comment by GMNGeoffrey** [comment]
Sure, I'm alright with that

**Comment by dcaballe** [comment]
Yeah, that sounds good to me. Version 12 is only one year behind and I would expect clang format to be relatively stable. Let's go with that!
